it/controllers/benefits/reimbursed/VouchersOrCreditCardsBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,534 @@
+
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+                Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlPost(pageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+                headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            "has the correct status" in {
+              result.status shouldBe BAD_REQUEST
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+            h1Check(user.specificExpectedResults.get.expectedHeading)
+            textOnPageCheck(expectedCaption, captionSelector)
+            textOnPageCheck(user.specificExpectedResults.get.paragraph, paragraphSelector)
+            radioButtonCheck(yesText, 1, Some(false))
+            radioButtonCheck(noText, 2, Some(false))
+            buttonCheck(expectedButtonText, continueButtonSelector)
+            formPostLinkCheck(continueLink, continueButtonFormSelector)
+            welshToggleCheck(user.isWelsh)
+
+            errorSummaryCheck(user.specificExpectedResults.get.expectedErrorText, Selectors.yesSelector)
+            errorAboveElementCheck(user.specificExpectedResults.get.expectedErrorText, Some("value"))
+          }
+        }
+      }
+    }
+
+    "redirect to another page when a valid request is made and then" should {
+
+      val user = UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedAgentEN))
+
+      "Update the vouchers or credit cards question to no, and vouchers or credit cards to none when the user chooses no, redirects to CYA if prior submission" which {
+
+        lazy val form: Map[String, String] = Map(YesNoForm.yesNo -> YesNoForm.no)
+
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = true, cyaModel("employerName", hmrc = true,
+            Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(pageUrl(taxYearEOY), body = form, follow = false, welsh = user.isWelsh,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "redirects to the check your details page" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY,employmentId).url)
+        }
+
+        "updates vouchers or credit cards question to no and vouchers or credit cards to none" in {
+          lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
```

Comment:
update to cyaModel to remove the warning

Reply:
Updated as requested.